apps/system/js/app_chrome.js

```diff
@@ -473,7 +477,27 @@
   };
 
   AppChrome.prototype.handleSecurityChanged = function(evt) {
-    this.title.dataset.ssl = this.app.getSSLState();
+    var state = this.title.dataset.ssl = this.app.getSSLState();
+    var isRTL = (document.documentElement.dir === 'rtl');
+    var showLockIcon = (state == 'secure' || state == 'broken');
+    var beforeIndent = 0;
+
+    if (!this.useCombinedChrome()) {
+      if (this.app.isPrivateBrowser()) {
+        beforeIndent += this.PRIVATE_ICON_WIDTH;
+      }
+      if (showLockIcon) {
+        beforeIndent += this.LOCK_ICON_WIDTH;
+      }
+      if (isRTL) {
+        // close button is currently on left in gaia-header in LTR & RTL
```

Comment:
note that once gaia-header supports RTL mode, titleStart will have a LTR/RTL meaning too  (left in LTR, right in RTL). This will make your life easier -- but we'll need to take care to this.

It's maybe a good time to file a bug for this (gaia-header supporting RTL), another bug to change this here, and add the bugs references here in a comment as a reminder.

I still curse the day someone decided to not make gaia-header RTL-aware.